sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/expressions.scala

```diff
@@ -479,12 +479,15 @@ object LikeSimplification extends Rule[LogicalPlan] {
   private val startsAndEndsWith = "([^_%]+)%([^_%]+)".r
   private val contains = "%([^_%]+)%".r
   private val equalTo = "([^_%]*)".r
+  private val endsWithUnderscore = "([^_%]+)_".r
+  private val startsWithUnderscore = "_([^_%]+)".r
+  private val startsAndEndsWithUnderscore = "([^_%]+)_([^_%]+)".r
 
   def apply(plan: LogicalPlan): LogicalPlan = plan transformAllExpressions {
     case Like(input, Literal(pattern, StringType)) =>
       if (pattern == null) {
         // If pattern is null, return null value directly, since "col like null" == null.
-        Literal(null, BooleanType)
+        Literal(false, BooleanType)
```

Comment:
why did you change this?

Reply:
I was planning on changing `like null` to `like false` so that we can change the underlying table read but I realized that `like null` is a valid check. I will revert this.